src/main/java/com/minecolonies/blockout/views/DropDownList.java

```diff
@@ -0,0 +1,308 @@
+package com.minecolonies.blockout.views;
+
+import com.minecolonies.blockout.controls.Button;
+import com.minecolonies.blockout.controls.ButtonVanilla;
+import com.minecolonies.blockout.controls.Label;
+import com.minecolonies.blockout.views.ScrollingList;
+import com.minecolonies.blockout.OverlayView;
+import com.minecolonies.blockout.Pane;
+import com.minecolonies.blockout.PaneParams;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * A DropDownList is a Button which when click display a ScrollingList below it.
+ */
+//TODO do not extend ButtonVanilla but extend view
+public class DropDownList extends ButtonVanilla implements Button.Handler
+{
+    /**
+     * View in which the list will be displayed.
+     */
+    protected OverlayView   overlay;
+
+    /**
+     * List to choose from.
+     */
+    protected ScrollingList list;
+    /**
+     * date required to fill the list.
+     */
+    protected DataProvider  dataProvider;
+
+    /**
+     * handler of the onSelectedItemChanged.
+     */
+    protected Handler       handler;
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER') "handler" is the name of a field in "Button". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2387)